src/main/java/com/artipie/Credentials.java

```diff
@@ -274,4 +290,47 @@ public FromEnv(final Map<String, String> env) {
             return CompletableFuture.completedFuture(new AuthFromEnv());
         }
     }
+
+    /**
+     * User.
+     * @since 0.11
+     */
+    final class User {
+
+        /**
+         * User name.
+         */
+        private final String uname;
+
+        /**
+         * User email.
+         */
+        private final String mail;
```

Comment:
@olenagerasimova not every user has an email, so I think it will make more sense to make it `Optional`. If the field is required in certain `Slice` then we may put default value there, but not in all places where `User` is added/modified